### PR TITLE
[NestedTensor] Integrate the layer normalization operator along the jagged dimension into NestedTensor

### PR DESCRIPTION
**Pull request by jananisriram**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #132172
* #132170

Modify the existing `layer normalization` operator in PyTorch, invoked by `torch.layer_norm`, to allow for reductions along the jagged dimension of a nested tensor. The function originally had a basic implementation for reducing along 1 non-ragged dimension. This diff, which uses the `aten` padding operator, enables PyTorch users to invoke `torch.nn.functional.layer_norm` on a nested tensor when reducing along the ragged dimension, e.g. `*` in a `(B, *, M)` or `(B, *, M, N)` nested tensor.

Write unit tests based on the `softmax` jagged operator to verify the accuracy of the ragged reduction implementation for `torch.nn.functional.layer_norm`. Add unit tests to verify error handling for unsupported features.

Note that this implementation is limited to nested tensors with `ragged_idx == 1`, i.e. the ragged dimension is not transposed. The layer normalization operator also requires an operation on a 2-dimensional layer; for nested tensors with 4 or more dimensions, I flatten the extra dimensions, then unflatten them after performing layer normalization.

### CODE REVIEW

**Review by davidberard98 — APPROVED** [review]
lgtm as long as tests pass!
